### PR TITLE
Embeddings: exclude files by size in `embedFiles`

### PR DESCRIPTION
**Pull request by camdencheek**
Currently, we exclude files by their size when we create the input list of files. The problem with this is it's pretty distant from the place we do all the other file skipping, so it makes collecting stats about why files were skipped more difficult. 

This updates the code to maintain the list of files along with their size so that we can add them to the excluded counts. If we do not list them, we cannot count them. This has the added bonus that we can keep track of the _sizes_ of files that were skipped, so we have a full bookkeeping of how much of the repo was indexed.

Stacked on #51018 

## Test plan

Existing tests. Updated tests to use new interfaces.

### CODE REVIEW

File: enterprise/cmd/worker/internal/embeddings/repo/handler.go [hunk 1]
```diff
@@ -90,3 +81,30 @@ func (h *handler) Handle(ctx context.Context, logger log.Logger, record *repoemb
...
+	gitserver gitserver.Client
+}
+
+func (r *revisionFetcher) Read(ctx context.Context, fileName string) ([]byte, error) {
+	return r.gitserver.ReadFile(ctx, nil, r.repo, r.revision, fileName)
+}
+
+func (r *revisionFetcher) List(ctx context.Context) ([]embed.FileEntry, error) {
+
+	fileInfos, err := r.gitserver.ReadDir(ctx, nil, r.repo, r.revision, "", true)
```
Comment (jtibshirani) [inline enterprise/cmd/worker/internal/embeddings/repo/handler.go:97]: It's nice we can just grab the file sizes here too. Is there any difference in what files will be loaded between this call and the old one below?

```
h.gitserverClient.ListFiles(ctx, nil, repo.Name, record.Revision, &protocol.ListFilesOpts{
		IncludeDirs:      false,
		MaxFileSizeBytes: maxFileSize,
	})
```

(Besides the max file size, which we're purposefully handling later)

**Review by jtibshirani — COMMENTED** [review]
This makes sense to me, just had one question. I was also a little surprised to see changes to `WORKSPACE`, maybe that's expected though?

File: enterprise/cmd/worker/internal/embeddings/repo/handler.go (hunk 1 above)
Comment (camdencheek) [reply enterprise/cmd/worker/internal/embeddings/repo/handler.go:96]: `gitserver.ListFiles` is just implemented as a [light wrapper around ReadDir](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@52a6cc30540f6692b681ef0e179212ad86fad722/-/blob/internal/gitserver/commands.go?L1028-1028).

But your comment made me remember that I forgot to include the logic to exclude directories. I just pushed a commit to fix that.

**Comment by camdencheek** [comment]
> surprised to see changes to WORKSPACE

I was also surprised. That was generated from `bazel run :gazelle -- fix`. Maybe we just bumped a Bazel version or something?

**Pull request merged**